### PR TITLE
Implement getUserPublicKey()

### PR DESCRIPTION
**Pull request by mahmoudsaeed**

### CODE REVIEW

**Comment by Bilal-S** [comment]
@vrabeshko 
can you review and provide comments if any please.

**Comment by mahmoudsaeed** [comment]
@vrabeshko I resolved the issues.

File: test/src/XcooBee/Core/Api/UsersTest.php [hunk 1]
```diff
@@ -87,8 +87,65 @@ public function testGetConversations_UseConfig()
...
+            'apiKey' => 'testapikey',
+            'apiSecret' => 'testapisecret'
+        ]);
+    }
+
+    public function testGetUserPublicKey_NoPgpFound()
+    {
+        $usersMock = $this->_getMock(\XcooBee\Core\Api\Users::class, [
+            '_request' => true,
+            'getUserPublicKey' => $this->_createResponse(200, (object) ['data' => []]),
```
Comment (vrabeshko) [inline test/src/XcooBee/Core/Api/UsersTest.php]: you mock the method first and then call it. In this case you have to mock only `_request` to return `Response` object with empty data. And in this case check that `getUserPublicKey` returns `null`

File: test/src/XcooBee/Core/Api/UsersTest.php [hunk 2]
```diff
@@ -87,8 +87,65 @@ public function testGetConversations_UseConfig()
...
+            '_request' => true,
+        ]);
+
+        $usersMock->expects($this->once())
+                ->method('_request')
+                ->will($this->returnCallback(function ($query, $params) {
+                            $this->assertEquals(['xid' => 'testXid'], $params);
+                        }));
+
+        $usersMock->getUserPublicKey('testXid');
```
Comment (vrabeshko) [inline test/src/XcooBee/Core/Api/UsersTest.php]: in this case mock `_request` to return valid response with data and `pgp_public_key` and compare response of `getUserPublicKey` to your mocked key

**Pull request merged**